include/openthread/ip6.h

```diff
@@ -438,6 +438,23 @@ otMessage *otIp6NewMessageFromBuffer(otInstance              *aInstance,
                                      uint16_t                 aDataLength,
                                      const otMessageSettings *aSettings);
 
+/**
+ * Checks whether the given message may be looped back in a case of a multicast destination address.
+ *
+ * @param[in]  aMessage A pointer to the message.
+ *
+ */
+bool otIp6IsMulticastLoopEnabled(otMessage *aMessage);
```

Comment:
Also suggest adding these in `message.h` and not in `ip6.h` and with `otMessage` prefix and not `otIp6` prefix:
- As  `otMessageIsMulticastLoopEnabled()` and `otMessageSetMulticastLoop()`.
- They can go next to existing `otMessageIsLoopbackToHostAllowed()` and `otMessageSetLoopbackToHostAllowed()` APIs in `message.h`
- So all related APIs setting flags on `message` are next to each other.